array.c

```diff
@@ -4373,65 +4373,81 @@ take_items(VALUE obj, long n)
 
 /*
  *  call-seq:
- *    array.zip(*other_arrays) -> new_array
- *    array.zip(*other_arrays) {|other_array| ... } -> nil
+ *    zip(*objects) -> new_array
+ *    zip(*objects) {|other_array| ... } -> nil
  *
- *  When no block given, returns a new +Array+ +new_array+ of size <tt>self.size</tt>
- *  whose elements are Arrays.
+ *  For an *object* in *objects* that is not an array,
+ *  forms the the "other array" as <tt>object.to_ary</tt>, if defined,
+ *  or as <tt>object.each.to_a</tt> otherwise.
```

Comment:
I don't think we should have this as the beginning of the documentation. The most common case is that `objects` are all arrays and converting these objects to arrays is the uncommon case.

Reply:
Recast.